src/scripts/Components/DocumentTitle.tsx

```diff
@@ -15,23 +14,32 @@ import { getMeta } from "@core/application";
  *     <h1>Page Title</h1>
  * </DocumentTitle>
  */
-export default class DocumentTitle extends React.Component {
-    static propTypes = {
-        title: PropTypes.string,
-        children: PropTypes.node,
-    }
-
-    componentDidMount() {
+export default class DocumentTitle extends React.Component<IProps> {
+    public componentDidMount() {
         document.title = this.getHeadTitle(this.props);
     }
 
-    componentWillUpdate(nextProps) {
+    public componentWillUpdate(nextProps: IProps) {
         document.title = this.getHeadTitle(nextProps);
     }
 
-    getHeadTitle(props) {
-        const siteTitle = getMeta('ui.siteName', '');
-        const parts = [];
+    public render() {
+        if (this.props.children) {
+            return this.props.children;
+        } else {
+            return <h1>{this.props.title}</h1>;
+        }
+    }
+
+    /**
+     * Calculate the status bar title from the props.
+     *
+     * @param {IProps} props The props used to calculate the title.
```

Comment:
We should probably skip the comment type annotations and just put them in the function declaration. `+    private getHeadTitle(props: IProps): string {`